estate/models/estate_property.py

```diff
@@ -0,0 +1,107 @@
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Available From', copy=False, default=fields.Date.add(fields.Date.today(), months=3))
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area (sqm)')
+    garden_orientation = fields.Selection(string='Garden orientation',
+                                          selection=[('north', 'North'),
+                                                    ('south', 'South'),
+                                                    ('east', 'East'),
+                                                    ('west', 'West')]
+                                          )
+    state = fields.Selection(string='State',
+                            selection=[('new', 'New'),
+                                    ('offer_received', 'Offer Received'),
+                                    ('offer_accepted', 'Offer Accepted'),
+                                    ('sold', 'Sold'),
+                                    ('cancelled', 'Cancelled')],
+                             default='new', required=True, copy=False)
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type')
+    property_tag_ids = fields.Many2many('estate.property.tag', string='Property Tag')
+    buyer_id = fields.Many2one('res.partner', string='Buyer', copy=False,
+                               domain=[('is_company', '=', False)])
+    salesperson_id = fields.Many2one('res.users', string='Salesperson',
+                                     default=lambda self: self.env.user)
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    total_area = fields.Integer('Total Area (sqm)', compute='_compute_total_area')
+    best_offer = fields.Float('Best Offer', compute='_compute_best_offer')
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)',
+        'The expected price must be strictly positive',
+    )
+
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price >= 0)',
+        'The selling price must be positive',
+    )
+
+    @api.depends('garden_area', 'living_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_offer = max(record.offer_ids.mapped('price'))
+            else:
+                record.best_offer = 0
```

Comment:
you can just do this
```suggestion
            record.best_offer = max(record.offer_ids.mapped('price'), default=0.0)
```